### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by kevin-10**
- Added property rewards gump to Huntsman Challenge
- POF no longer works on jewels
- Added propery functionality to water barrels/fillable barrels
- fixed crash fix

### CODE REVIEW

File: Scripts/Items/Consumables/WaterBarrel.cs [hunk 1]
```diff
@@ -12,7 +13,7 @@ public WaterBarrel()
 
         [Constructable]
         public WaterBarrel(bool filled)
```
Comment (Argalep) [inline Scripts/Items/Consumables/WaterBarrel.cs]: Water barrel can not movable.

File: Scripts/Items/Consumables/BaseWaterContainer.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@ public abstract class BaseWaterContainer : Container, IHasQuantity
         public BaseWaterContainer(int Item_Id, bool filled)
             : base(Item_Id)
         {
-            this.m_Quantity = (filled) ? this.MaxQuantity : 0;
+            m_Quantity = (filled) ? MaxQuantity : 0;
```
Comment (Argalep) [inline Scripts/Items/Consumables/BaseWaterContainer.cs]: m_Quantity change to Quantity

File: Scripts/Items/Consumables/WaterBarrel.cs (hunk 1 above)
Comment (kevin-10) [inline Scripts/Items/Consumables/WaterBarrel.cs]: They will be marked as non-movable when they are placed by filling the fillable barrel.

File: Scripts/Items/Consumables/BaseWaterContainer.cs (hunk 2 above)
Comment (kevin-10) [inline Scripts/Items/Consumables/BaseWaterContainer.cs]: I don't understand what you're saying here.

**Pull request merged**